### DISCUSSION TITLE
How to clone loaded lists with PnPContext?

### DISCUSSION THREAD

**Discussion by heinrich-ulbricht** [pnp/pnpcore#1388] [Q&A]
I do a lot of context cloning. Partially out of historic reasons (had issue when not cloning, and cloning was the solution), partially when doing parallel calls that should not interfere with each other.

Now I need certain lists to be loaded in the `PnPContext`, e.g. the Site Pages library. It should be present in all clones and is currently loaded over and over again, since loaded lists are not cloned.

I found this comment in the code:

                // Copy the additional initialization properties (if any)
                // Only "basic" properties are cloned, complex properties (e.g. Web.Lists) are not cloned as 
                // a common use case of using a cloned context is to start "clean", basic web/site properties do not change 
                // when cloning for the same site url, but one might want to load collections (= complex property)
                // using a different filter

But I would _love_ the lists to be cloned.

My question is: is cloning loaded lists (e.g. Pages library) possible from a technical point of view? And since I have no clue how that would work: how would I start implementing that (architectural big picture)?

**Comment by jansenbe** [comment]
@heinrich-ulbricht : cloning of a loaded list technically is possible, but there's quite a lot of complexity involved. We today do some cloning alike things in `PnPContextFactory.CopyContextInitializationAsync` and `EntityManager.ReplicateParentHierarchy` which can serve as inspiration.  You can't code this outside of PnP Core SDK as there are internal properties that have to be set as part of the cloning. I'm open for ideas on a PR that will enable this.

**Reply by heinrich-ulbricht** [reply]
Thanks for the pointers, I'll have a look if I can understand that. I'm working on a forked version of PnP.Core, so I'm quick with testing.
